hastory-cli/src/Hastory/Cli/Commands/Sync.hs

```diff
@@ -55,22 +55,20 @@ updateOrInsert serverEntity = do
 -- return entries that are unknown to the client.
 getMaxSyncWitness :: (MonadReader Settings m, MonadUnliftIO m) => m Int64
 getMaxSyncWitness = do
-  mEntityEntry <- runDb' (selectFirst [EntrySyncWitness !=. Nothing] [Desc EntrySyncWitness])
-  case mEntityEntry of
+  mEntityEntry <- runDb' $ selectFirst [EntrySyncWitness !=. Nothing] [Desc EntrySyncWitness]
+  let mSyncWitness = entrySyncWitness . entityVal =<< mEntityEntry
+  case mSyncWitness of
     Nothing -> pure 0
```

Comment:
You're still using `pure` in both branches here. You can put the `pure $ ` before the `case`.